src/Compatibility/Core/src/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -159,9 +162,9 @@ public override void ViewDidDisappear(bool animated)
 			PageController.SendDisappearing();
 		}
 
-		public override void ViewDidLayoutSubviews()
+		public override void ViewWillLayoutSubviews()
```

Comment:
I moved pretty much all of  this `Frame` code into the `ParentingViewController`. The `ParentingViewController` knows what size frame it wants based on the offset from the navbar so it seemed to work better in the flow of things to set offsets in there